src/runtime/virtcontainers/sandbox.go

```diff
@@ -670,30 +671,30 @@ func (s *Sandbox) createCgroups() error {
 	// Depending on the SandboxCgroupOnly value, this cgroup
 	// will either hold all the pod threads (SandboxCgroupOnly is true)
 	// or only the virtual CPU ones (SandboxCgroupOnly is false).
-	s.sandboxCgroup, err = cgroups.NewSandboxCgroup(cgroupPath, &resources, s.config.SandboxCgroupOnly)
+	s.sandboxController, err = resCtrl.NewSandboxResourceController(cgroupPath, &resources, s.config.SandboxCgroupOnly)
 	if err != nil {
 		return fmt.Errorf("Could not create the sandbox cgroup %v", err)
 	}
 
 	// Now that the sandbox cgroup is created, we can set the state cgroup root paths.
-	s.state.SandboxCgroupPath = s.sandboxCgroup.Path()
+	s.state.SandboxCgroupPath = s.sandboxController.ID()
 	s.state.OverheadCgroupPath = ""
 
 	if s.config.SandboxCgroupOnly {
-		s.overheadCgroup = nil
+		s.overheadController = nil
 	} else {
 		// The shim configuration is requesting that we do not put all threads
 		// into the sandbox cgroup.
 		// We're creating an overhead cgroup, with no constraints. Everything but
```

Comment:
while we're updated from cgroups to a more generic `resource-controller` interface, I do note that most of the comments (and some naming) at the sandbox level are about cgroups.  Since we don't have any other implementations of resource controller introduced to Kata at this point, perhaps we can address this part at a later time, but there'll be a disconnect here?

Reply:
All those are good points.
So I added a `Type` method to the `ResourceController` interface, and tried to make a lot of the `sandbox.go` comments and error strings less cgroups specific. Let me know if it looks better to you.